tests/istio.io/README.md

```diff
@@ -6,12 +6,130 @@ This folder contains tests for the different examples on Istio.io.
+    Run()
+}
+```
+
+This method allows a test to wait for a set of pods (specified by the fetch function) to start, or for 30 seconds, whichever comes first.
+
+To write a test:
+
+1. Add the following imports to your file:
+
+```golang
+"istio.io/istio/pkg/test/framework"
+"istio.io/istio/pkg/test/framework/components/environment"
+"istio.io/istio/pkg/test/framework/components/istio"
+"istio.io/istio/pkg/test/istio.io/examples"
+```
+
+2. Next, create a function called `TestMain` as follows. This function will setup the Istio environment used by the tests. The `istio.Setup` function accepts an optional method to customize the Istio environment deployed.
+
+```golang
+func TestMain(m *testing.M) {
+  framework.NewSuite("mtls-migration", m).
+    SetupOnEnv(environment.Kube, istio.Setup(&ist, nil)).
+    RequireEnvironment(environment.Kube).
+    Run()
+}
+```
+
+3. After creating `TestMain`, tests can be created as follows. In this example, the NewMultiPodFetch function is waiting for all pods in the foo, bar,and legacy namespaces to start. The GetCurlVerifier method is verify the responses from the curl script. Finally, the run method triggers the test to run.
+
+```golang
+func TestMTLS(t *testing.T) {
+  examples.New(t, "mtls-migration").
+    RunScript("create-ns-foo-bar-legacy.sh", examples.TextOutput, nil).
+    WaitForPods(examples.NewMultiPodFetch("foo")).
+    RunScript("curl-foo-bar-legacy.sh", examples.TextOutput, examples.GetCurlVerifier([]string{"200", "200", "200"})).
+    Run()
+}
+```
+
+### Built in Verifiers
+
+The framework includes some predefined verifiers for scripts. These are described below.
+
+* `examples.GetCurlVerifier`: The GetCurlVerifier method is used to create a verifier that parses the output of a curl script and compares each response. If a response of `000` is returned from curl, the verifier will compare the next line against the provided output. Finally, the run method triggers the test to run. This validator expects responses in the form `sleep.foo to httpbin.foo: 200` except for when the response code is 000, in which case, the next line of output is expected to be an error statement. For example, the following output can be verified with `examples.GetCurlVerifier([]string{"200", "200", "000", "command terminated with exit code 56"}))`.
+
+```bash
+sleep.foo to httpbin.foo: 200
+sleep.bar to httpbin.foo: 200
+sleep.legacy to httpbin.foo: 000
+command terminated with exit code 56
+``` 
+
+
+
+### Output
+
+When the test framework runs, it creates a directory called output. This directory contains a copy of all YAML files, scripts, and a file for each script containing the output.
+
 ## Executing Tests Locally using Minikube
 
 * Create a Minikube environment as advised on istio.io
+* Tests can be run using the below command. 
 
-* Run the tests as follows:
+```bash
+ go test ./... -p 1 --istio.test.env kube --istio.test.kube.config ~/.kube/config -v -istio.test.ci
```

Comment:
Is the `--istio.test.kube.config ~/.kube/config` actually needed here since I believe it's a default? I'm fine with it there if it would be commonplace for someone to change it to something else.

Reply:
Yeah, just tested and this runs without --istio.test.kube.config or -istio-test.ci. Will remove both. They're explained on the linked parameters.